emperor/support_files/js/draw.js

```diff
@@ -295,11 +299,80 @@ define(['underscore', 'three', 'jquery'], function(_, THREE, $) {
     // the line will contain the two vertices and the described material
     // we increase the number of points to have a smoother transition on
     // edges i. e. where the trajectory changes the direction it is going
-    lineGeometry = new THREE.TubeGeometry(path, (points.length - 1) * 3, radius,
-                                          10, false);
+    lineGeometry = new THREE.TubeGeometry(path, (points.length - 1) * NUM_TUBE_SEGMENTS, radius,
+                                          NUM_TUBE_CROSS_SECTION_POINTS, false);
+
+    return new THREE.Mesh(lineGeometry, material);
+  }
+  
+  function drawTrajectoryLineDynamic(trajectory, currentFrame, color, radius) {
+    // based on the example described in:
+    // https://github.com/mrdoob/three.js/wiki/Drawing-lines
+    var material, points = [], lineGeometry, limit = 0, path;
+
+    _trajectory = trajectory.representativeInterpolatedCoordinatesAtIndex(currentFrame);
+    if (_trajectory === null || _trajectory.length == 0)
+      return null;
+
+    material = new THREE.MeshPhongMaterial({color: color});
+    material.matrixAutoUpdate = true;
```

Comment:
Just to confirm, do these parameters need to be set after creation? In other words, can these be `material = new THREE.MeshPhongMaterial({color: color, matrixAutoUpdate: true, transparent: false});`. I checked the documentation and couldn't confirm ...

Reply:
As far as I can tell in the current implementation, the difference for boolean flags is only stylistic, but it heavily depends on the data types of the values of the map.

The map passed to the THREE.MeshPhongMaterial is named "parameters" in the MeshPhongMaterial constructor.  The final line of this constructor calls a function of Material called "setValues".  This setValues call loops over keys, warns about keys that it knows are deprecated, and defensively copies certain data types.  It's thus probably more future proof to use their setValues mechanism.  I'll make the change here, but I think updating existing code for consistency with this pattern is out of the scope of this PR.

Reply:
Also, matrixAutoUpdate is not a member of THREE.MeshPhongMaterial, but of the Mesh object itself.  Unclear what was intended or if this property of meshes needs to be set (True by default?  When would this property make a difference?).  Since setValues alerts the developer to this, I've updated my methods to use the constructor as you suggested, and removed the matrixAutoUpdate: true.

Reply:
Done for the new methods I've added.  Should discuss using this convention separately in the rest of the codebase as a separate item.